holoplot/converter.py

```diff
@@ -0,0 +1,652 @@
+from __future__ import absolute_import
+from functools import partial
+from distutils.version import LooseVersion
+
+import param
+import holoviews as hv
+import pandas as pd
+
+from holoviews.core.spaces import DynamicMap, Callable
+from holoviews.core.overlay import NdOverlay
+from holoviews.core.layout import NdLayout
+from holoviews.element import (
+    Curve, Scatter, Area, Bars, BoxWhisker, Dataset, Distribution,
+    Table, HeatMap, Image, HexTiles, QuadMesh
+)
+from holoviews.operation import histogram
+from holoviews.streams import Buffer, Pipe
+
+try:
+    import xarray as xr
+except ImportError:
+    xr = None
+
+try:
+    import streamz.dataframe as sdf
+except ImportError:
+    sdf = None
+
+try:
+    from intake.source.base import DataSource
+except ImportError:
+    DataSource = None
+
+try:
+    import dask.dataframe as dd
+except ImportError:
+    dd = None
+
```

Comment:
I'm alarmed by seeing all these conditional imports here, because it means that if I have any of these packages in my environment, anything I use with holoplot now depends on them to be working without errors.  Plus this approach will slow down everyone on every usage, and more so when more libraries are added.

Given that this is a package very likely to be used with only one of the packages from this list by any given user at any given time, shouldn't the imports be postponed until later and only either when the functionality is actually used or else swallowing all exceptions, not just ImportError for any library not used?

Reply:
Yes, that's one of my to-do items but atm it's somewhat pointless because HoloViews will import all these things anyway (except for streamz maybe). The solution I come up with here can then presumably should also be applied to HoloViews ideally.

Reply:
Ok, good luck!

Reply:
If you need any extra motivation, this issue was theoretical when I wrote the above, but now that I actually tried it, it's immediately a problem:

![image](https://user-images.githubusercontent.com/1695496/39452157-dfbffec6-4c96-11e8-9bfe-5c33040b2f3e.png)

I.e. I presumably have an older streamz library installed, and even though this example doesn't use streamz at all, it's failing due to streamz anyway.